spot_rl_experiments/spot_rl/utils/search_table_location.py

```diff
@@ -0,0 +1,517 @@
+import os
+import time
+from copy import deepcopy
+from glob import glob
+from typing import Any
+
+import cv2
+import magnum as mn
+import numpy as np
+import open3d as o3d
+import zmq
+from spot_rl.envs.base_env import SpotBaseEnv
+from spot_rl.utils.heuristic_nav import get_3d_point
+from spot_rl.utils.plane_detection import plane_detect
+
+# from spot_rl.utils.pose_correction import (
+#     detect,
+#     generate_point_cloud,
+#     load_model,
+#     segment,
+# )
+from spot_wrapper.spot import Spot, image_response_to_cv2
+
+
+def area(x1, y1, x2, y2):
+    return np.abs(y2 - y1) * np.abs(x2 - x1)
+
+
+def select_the_bbox_closer_to_camera(predictions, raw_depth, camera_intrinsics):
+    closer_dist = np.inf
+    # upper_dist = -np.inf
+    closer_i = 0
+    h, w = raw_depth.shape
+    for i, prediction in enumerate(predictions):
```

Comment:
Add a comment here, something like iterating all predictions to find the location